Solutions/CiscoUmbrella/Data Connectors/ciscoUmbrellaDataConn/__init__.py

```diff
@@ -445,6 +448,33 @@ def parse_csv_cdfw(self, csv_file):
                 event['EventType'] = 'cloudfirewalllogs'
                 yield event
 
+    def parse_csv_fw(self, csv_file):
+        csv_reader = csv.reader(csv_file.split('\n'), delimiter=',')
+        for row in csv_reader:
+            if len(row) > 1:
+                if len(row) >= 14:
```

Comment:
I believe, you would require minimum 30 columns to work below parsing. Why 14 ? I hope you are confident about the position number, maximum index you use is 29 (position 30)

Reply:
Yes Anki,Checked it ,Will correct it its parsing only 14 rows earlier then will cross check entire code,noticed schema is incorrect in cloud firewall logs [Log Formats and Versioning (umbrella.com)](https://docs.umbrella.com/umbrella-user-guide/docs/log-format-and-versioning#firewall)
Verdict is not present and its Action now ,will change the existing 14 rows to 30 and Verdict to Action

Reply:
As discussed, reverted the changes, as we are not sure the position of this column as it's not clear in documentation, will do once get confirmation from cisco support team.